packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js

```diff
@@ -15,25 +27,205 @@ const i18n = {
+            fContactPreferencesModule: {
+                state,
+                actions,
+                namespaced: true
+            }
+        }
+    }),
+    mocks
+});
+
+describe('ContactPreferences Component', () => {
+    beforeEach(() => {
+        // Arrange & Act
+        dataDefaults = () => ({
+            isFormDirty: false,
+            showErrorPage: false
+        });
+        cookiesSpy = jest.fn();
+        httpSpy = jest.fn();
+        sutMocks = {
+            $parent: {
+                $emit: jest.fn()
+            },
+            $http: httpSpy,
+            $cookies: cookiesSpy
+        };
+        sutProps = {
+            authToken: token,
+            smartGatewayBaseUrl: baseUrl
         };
+        wrapper = mountContactPreferences();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('when mounting the component and calling `initialise`', () => {
+        it('should call the load action with the correct parameters', () => {
+            // Assert
+            expect(storeActions.loadPreferences).toHaveBeenCalledWith(expect.any(Object), {
+                api: {
+                    baseUrl,
+                    cookies: cookiesSpy,
+                    httpClient: httpSpy
+                },
+                authToken: token
+            });
+        });
+
+        it('should set showErrorPage flag to true if an error occurs', () => {
+            // Arrange
+            const errorActions = {
+                loadPreferences: jest.fn().mockImplementationOnce(() => {
+                    throw new Error('some-error');
+                })
+            };
+            wrapper = mountContactPreferences({ actions: errorActions });
+
+            // Assert
+            expect(wrapper.vm.showErrorPage).toEqual(true);
+        });
+
+        it('should not show the error card if no errors', () => {
+            // Arrange
+            const element = wrapper.find('[data-test-id="contactPreferences-error-card"]');
+
+            // Assert
+            expect(element.exists()).toEqual(false);
+        });
+
+        it('should show the error card if showErrorFlag is true', async () => {
+            // Arrange
+            await wrapper.setData({ showErrorPage: true });
+
+            // Act
+            const element = wrapper.find('[data-test-id="contactPreferences-error-card"]');
+
+            // Assert
+            expect(element.exists()).toEqual(true);
+        });
 
-        // Act
-        const wrapper = shallowMount(ContactPreferences, {
-            i18n,
-            localVue,
-            propsData
+        it('should set the `isFormDirty` flag to false', async () => {
+            // Arrange
+            await wrapper.setData({ isFormDirty: true });
+            const expected = false;
+
+            // Act - re-mount the page
+            wrapper = mountContactPreferences();
+
+            // Assert
+            expect(wrapper.vm.isFormDirty).toEqual(expected);
```

Comment:
Not sure the variable adds much here
```suggestion
            expect(wrapper.vm.isFormDirty).toEqual(false);
```